### PR TITLE
Testing branch

### PR DESCRIPTION
**Pull request by Redvanisation**

### CODE REVIEW

**Comment by maelfosso** [comment]
Hello coders,

Thank you for that great code you write, all the test passed. Before moving forward, there are a few things we should see first

- [x] Why is there no testing for the **Player** class?

- [x] When a method depends on another method (f1 calls f2), it's preferable to test the independent method (f2) before the other one (f1). `add_piece` use `piece_location_valid?` you should write a test for `piece_location_valid?` before the test of `add_piece`. 

- [x] About `piece_location_valid?`, I think you should write a test with **nil** pass in parameters.

- [x] About `valid_coords?`, I think you should write a test with **nil** pass in parameters. It's because of his first condition `if !coords`

- [x] I don't think that `add_piece` was well tested. You should expect at the end whether or not the piece added is at the coordinates sent to the method.

- [x] As you decide to write a test for `coords_available?`, please show us how the expected to result in both cases `true` and `false`.

- [x] About the **Game** class, there is not a declaration from a Game instance. If you're writing some tests from this class, how are you going to test its method without having any instances?

- [x] About the **Game** please you should rewrite all the tests. You could not write a test of that class and the `expect` method is always receiving method from **board**, it must only receive method from the **Game** class as is that class you're testing, you should expect something from methods from the **Game** class only. Check the comment I put above.

Please, make those changes and submit a new code review request.

File: spec/game_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    let(:player2) {Player.new("player2", "O")}
+    let(:board) {Board.new}
+    let(:piece) { "x" }
+    let(:piece2) { "o" }
+
+    describe "#check_draw" do
+
+        it "returns true and a message in case of draw and false in the opposite" do
+            allow(board).to receive(:full?).and_return(true)
+            expect(board.full?).to eq(true)
```
Comment (maelfosso) [inline spec/game_spec.rb]: You're testing the **Game** class, not the **Board** one. 
So I think you should rather have
`expect(game.check_draw).to eq(true)`

**Comment by mcrd25** [comment]
# This project is currently being reviewed by TSE [Maya Douglas](@mcrd25)

File: .rspec [hunk 2]
```diff
@@ -0,0 +1,2 @@
+--require spec_helper
+--format documentation
```
Comment (mcrd25) [inline .rspec:2]: 👍

File: spec/game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    let(:piece) { "x" }
+    let(:piece2) { "o" }
+
+    describe "#check_draw" do
+
+        it "returns true and a message in case of draw and false in the opposite" do
+            
+            allow(board).to receive(:full?).and_return(true)
+
+            expect(game.check_draw).to eq(true)
```
Comment (mcrd25) [inline spec/game_spec.rb:42]: This test checks for a positive scenario, but you also should have a test that checks when you expect draw to equal false

File: spec/game_spec.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        end
+    end
+
+    describe "#check_victory" do
```
Comment (mcrd25) [inline spec/game_spec.rb:53]: Great job implementing for negative and positive scenario!

File: spec/game_spec.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        it "Returns false when the there is no winning color combination in the rows" do
+
+            allow(board).to receive(:winning_combination?).and_return(false)
+
+            expect(game.check_victory).to eq(false)
+        end
+
+    end
+
+    describe "#check_game_over" do
```
Comment (mcrd25) [inline spec/game_spec.rb:71]: Again, good job!

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Wow! the project looks neat! Really good job! Almost time for blast off 🚀 

Hi, @Redvanisation  and @VihoSerge 👋,

Hope all is well, you've done a good job, however, I am wondering why @Redvanisation  has 26 of the 29 commits while @VihoSerge  only has 3. I'm sure there is a reasonable explanation for this 😅 

Please also request re-review for this project under Ruby Testing not OOP.

You guys have done a great job, however not all requirements to pass have been met, mainly because not all of your public methods have been tested 😢.

**SEE CHECKLIST BELOW**

# Minimum Requirements Checklist

To consider the project 'complete' it must satisfy the following points:

### Project Specification 

- [x] All the public methods of the classes implemented in project OOP are tested
- [x] Check if specs are passing on your machine.
- [x] Check if specs are failing if you break some code in tested class.

### Crucial things to make project works

- [x] The project should be done solo or collaboratively and that the commit history reflects
- [x] The project has a descriptive README.md

- [x] code is clean: indentation with two spaces, line lengths, spaces around operators 
- [x] Ensure that students work with Git Flow: production `master` branch + main `develop` branch + Pull Requests from feature branches to the `develop`
- [x] Specs cover positive and negative scenarios 
      Example: there is a spec for correct usage of method [e.g. “it “creates object if all required attributes are passed”] 
      and also there is spec for wrong usage of method  [e.g. “it “does not create object if required attributes are missing”]. 
*For most part you've passed this except for that one method in your game class 😢*
- [x] Specs for each method should be wrapped in one `describe` block. 
- [x] You should have one `it` block for each expected behaviour.
- [x] Instead of using `before_each` to define a variable for the test cases, students should [Use `let` instead](https://github.com/rubocop-hq/rspec-style-guide#let-blocks).

I am really impressed with your projects. If you need any ideas of how to test your untested methods feel free to contact me.

***If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/)

**Comment by maelfosso** [comment]
# Good job

You can submit the projet completion form.

**Review by JamesOkunlade — APPROVED** [review]
Great job with the testing project.

Your work is good and meets all basic requirement.

It's high time you moved on to another one. Good luck with your future projects. :+1:

**Comment by Redvanisation** [comment]
Thanks for the reviews!

**Pull request merged**